src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { type FC } from 'react';
+
+import searchIcon from './assets/search-icon.svg';
+import { DEFAULT_PLACEHOLDER } from './SearchBar.constants';
+import { useSearchBar } from './SearchBar.hook';
+import { SearchContainer, StyledInput, SearchIconButton } from './SearchBar.styles';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const SearchBar: FC<SearchBarProps> = (props) => {
+  const {
+    placeholder = DEFAULT_PLACEHOLDER,
+    inputOnChange,
```

Comment:
The component accepts both `inputOnChange` and `onChange` props. This duplicates responsibility and conflicts with the common convention where `onChange` is the single handler for input changes. It also mismatches the types file duplication.

**Sugerencia:**
```typescript
Unify the API: remove `inputOnChange` and use `onChange` consistently, or if you need both give them clearer names and document their responsibilities. Update the hook and tests accordingly.
```